### PR TITLE
chore: Code cleanup in DatabaseModal

### PR DESCRIPTION
**Pull request by lyndsiWilliams**
### SUMMARY
Some of my cleanup changes were overwritten during a recent rebase, this reintroduces those changes.

### ADDITIONAL INFORMATION
- [ ] Has associated issue: [Clubhouse Ticket](https://app.clubhouse.io/preset/story/8020/clean-up-database-modal-rebase)
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal.tsx [hunk 1]
```diff
@@ -195,15 +195,12 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
...
-      sqlalchemy_uri: db ? db.sqlalchemy_uri : '',
-      database_name:
-        db && db.database_name.trim().length
-          ? db.database_name.trim()
-          : undefined,
-      impersonate_user: db ? db.impersonate_user || undefined : undefined,
-      extra: db && db.extra && db.extra.length ? db.extra : undefined,
-      encrypted_extra: db ? db.encrypted_extra || undefined : undefined,
-      server_cert: db ? db.server_cert || undefined : undefined,
+      sqlalchemy_uri: db?.sqlalchemy_uri || '',
```
Comment (AAfghahi) [inline superset-frontend/src/views/CRUD/data/database/DatabaseModal.tsx:198]: this looks so much cleaner.

**Review by AAfghahi — APPROVED** [review]
Why are you changing package-lock.json? Was that intentional?

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal.tsx (hunk 1 above)
Comment (lyndsiWilliams) [reply superset-frontend/src/views/CRUD/data/database/DatabaseModal.tsx:198]: Thank you!

**Comment by lyndsiWilliams** [comment]
> Why are you changing package-lock.json? Was that intentional?

It changed when I ran `npm ci`, so I added it. In the past I've encountered merge conflicts with the package-lock if this happens and I don't add the change.

**Review by betodealmeida — APPROVED** [review]

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal.tsx [hunk 2]
```diff
@@ -296,12 +293,7 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
   };
 
   const validate = () => {
-    if (
-      db &&
-      db.database_name.trim().length &&
-      db.sqlalchemy_uri &&
-      db.sqlalchemy_uri.length
-    ) {
+    if (db?.database_name?.trim().length && db?.sqlalchemy_uri?.length) {
```
Comment (eschutho) [inline superset-frontend/src/views/CRUD/data/database/DatabaseModal.tsx:296]: I think you can remove the lengths here. 0 and "" are both falsy.

**Comment by eschutho** [comment]
> > Why are you changing package-lock.json? Was that intentional?
> 
> It changed when I ran `npm ci`, so I added it. In the past I've encountered merge conflicts with the package-lock if this happens and I don't add the change.

You should be able to remove it. You shouldn't have to update package-lock unless you make changes to package.json.

**Comment by eschutho** [comment]
This looks great!

**Review by eschutho — CHANGES_REQUESTED** [review]
This looks great! One more small suggestion.

**Review by eschutho — APPROVED** [review]
Looks good!

**Comment by eschutho** [comment]
🏷 ready-to-merge

**Pull request merged**